src/Microsoft.Azure.SignalR/Internals/DefaultHubMessageSerializer.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Microsoft.Azure.SignalR
+{
+    // Copied from https://github.com/aspnet/AspNetCore/commit/3d93e095dbf2297fabf595099341c4cce673f32d
+    // Planning on replacing in the 5.0 timeframe
+    internal class DefaultHubMessageSerializer
+    {
+        private readonly List<IHubProtocol> _hubProtocols = new List<IHubProtocol>();
+
+        public DefaultHubMessageSerializer(IHubProtocolResolver hubProtocolResolver, IList<string> globalSupportedProtocols, IList<string> hubSupportedProtocols)
+        {
+            var supportedProtocols = hubSupportedProtocols ?? globalSupportedProtocols ?? Array.Empty<string>();
+            foreach (var protocolName in supportedProtocols)
+            {
+                // blazorpack is meant to only be used by the ComponentHub
+                // We remove it from the list here except when the Hub has a single protocol that is "blazorpack" because
+                // that identifies it as the ComponentHub
+                if (supportedProtocols.Count > 1 && string.Equals(protocolName, "blazorpack", StringComparison.OrdinalIgnoreCase))
```

Comment:
😮

Reply:
Yeah, it's a bit of a hack :). We want something that allows the Azure SignalR SDK to work without having to wait for any ASP.NET Core release though (and will work with existing 3.0 apps).

Reply:
😢